packages/console/src/pages/SignInExperience/components/ManageLanguageModal/LanguageEditor.tsx

```diff
@@ -72,11 +77,16 @@ const LanguageEditor = ({ selectedLanguageTag, onFormStateChange }: LanguageEdit
       .json<CustomPhraseResponse>();
 
     void mutate(updatedCustomPhrase);
+    onUpdated(updatedCustomPhrase);
```

Comment:
Looks like a side effect of mutate(updatedCustomPhrase). is there a way we could subscribe the to mutation instead of calling an onUpdated callback method separately?  That is goona have two sources of data flow.

Reply:
- `void mutate(updatedCustomPhrase)` is used to update the the `CustomPhraseResponse` data from `GET /custom-phrase/:languageKey`.

- `onUpdated(updatedCustomPhrase)` is used to update the the `Array<CustomPhraseResponse>` data from `GET /custom-phrase`.

Maybe extracting the data management methods into a hook is better.
I will update it later.

Reply:
Add an `use-custom-phrases` hook to manage the custom phrases list.

Reply:
Refactored with custom phrases context.